src/beacon/app/device_performance_endpoint.py

```
@@ -218,578 +405,359 @@
-    # First entry - the most recent maintenance or installation
-    if random.random() < 0.7:  # 70% chance to have maintenance record
-        days_ago = random.randint(days//4, days//2) if days > 4 else 1
-        maint_date = (end_date - timedelta(days=days_ago)).strftime("%Y-%m-%d")
-        maintenance_history.append({
-            "date": maint_date,
-            "type": "Routine",
-            "description": "Regular sensor calibration and cleaning"
-        })
+    # Calculate trends for each metric
+    trends = {
+        "data_completeness_trend": calculate_metric_trend(weekly_data, "completeness"),
```

Ruff (0.12.2)

Undefined name `calculate_metric_trend`

(F821)